core/backend/src/workspace/Workspace.ts

```diff
@@ -766,9 +990,144 @@ class WorkspaceDbImpl implements WorkspaceDb {
       throw new Error("no cloud container to prefetch");
     return CloudSqlite.startCloudPrefetch(cloudContainer, this.dbFileName, opts);
   }
+
+  public queryResource(search: WorkspaceResource.Search, callback: (result: WorkspaceResource.SearchResult) => void | "stop"): void {
+    this.withOpenDb((db) => {
+      db.withSqliteStatement(`SELECT value from ${search.resourceType}s WHERE id ${search.nameCompare ?? "="} ?`, (stmt) => {
+        stmt.bindString(1, search.nameSearch);
+        while (DbResult.BE_SQLITE_ROW === stmt.step()) {
+          if (callback({ workspaceDb: this, rscName: stmt.getValueString(0) }) === "stop")
```

Comment:
what if i want the name of the resource as well?

Reply:
good point, you will need that for anything other than "=".

Reply:
see tests

Reply:
@williamkbentley

Reply:
Similar question - I think it might be helpful to know what priority a resource has. 

So in the LandscapePro example, if there was a default trees db, and an admin configured DogwoodDb, it might be helpful to show the user a list of all trees grouped by source. I know the list of resources comes sorted by priority, but I dont see how the application can tell which priority the resource has.

Reply:
Resources don't directly have priority, the dictionaries that point to the WorkspaceDbs containing them do. You can use `WorkspaceDb.getWorkspaceDbs` to resolve a setting name to a list of dbs. You can pass a DbListFilter to it, which will supply to you for each db the cloud props and the dictionary from which it was resolved. The dictionary has a priority.